Development

moggerzfroggerz opened the pull request.

moggerzfroggerz committed:
First commitment of Dungeon Explorer

This has the ability for the player to eat, attack the monster, pick up a key item, check backpack/inventory.

moggerzfroggerz committed:
Update Game.cs

Added the option for the user to use the key to escape the dungeon, they are given the choice of two doors.

moggerzfroggerz committed:
Fixed monster heal

Monster regains health when the wrong door is chosen, fixed heal monster function.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
     {
         private Player player;
         private Room currentRoom;
+        private Monster monster;
```
The use of PascalCase and camelCase on classes and variables ensures the code is easy to read.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
-
+            currentRoom = new Room("You enter a dim room and are faced with a huge slug-like monster! You notice a shimmering item on the floor...");
+            player = new Player("Player", 100);
+            monster = new Monster("ScarySlug", 50);
```
The short and succinct description is attention grabbing and immediately throws the user into the action.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
+                    else
+                    {
+                        Console.WriteLine("Please input a letter displayed above.");
+                    }
```
Good implementation of a catch all clause to ensure that users don't enter the wrong key.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
     }
-}
\ No newline at end of file
+}
```
The addition of comments throughout would make it easier to follow what the code does.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
+                    Console.WriteLine($"Your current health is: {player.showHealth()}");
+                    Console.WriteLine($"Scary slug current health is: {monster.getHealth()}");
+                    string input = this.ExplorerInput();
+                    if (input == ("d"))
```
The use of single key inputs allow the user to more easily navigate over having to type entire instructions.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
+                                Console.WriteLine("Regardless of how hard you try, the key does not fit and the door will not open.");
+                                Console.WriteLine("In the time it took you to try the door, the scary slug regained 10 health.");
+                                monster.Heal();
+                                Console.WriteLine(monster.getHealth());
```
Minor aesthetic adjustment, you could have a display showing "Monster health: " Then the health, rather than just the number to ensure the user understands what the number means.

Winterspear commented on Game.cs:
```diff
@@ -1,26 +1,114 @@
+                        player.FindItems(PlayerInv);
+                        Console.WriteLine($"On the floor, there was {PlayerInv} and you picked it up.");
+                    }
+                    else if (input == ("b") && player.FindItems() == true)
```
It may be worth separating the conditions in this if statement into separate if statements as currently if the player does not have an item and enters b, the code requests them to enter a letter shown above still, even though they did.

moggerzfroggerz committed:
Acting on Anthony  and Jasmine's reviews

I acted on the reviews I received in the peer reviews. This included changing the function names to Pascal Case for consistency, as well as adding more comments to my code. I also changed the results of opening door 1.
